src/main/java/hudson/remoting/ClassFilter.java

```diff
@@ -89,6 +89,17 @@ public final Class check(Class c) {
     public static final ClassFilter DEFAULT = createDefaultInstance();
 
     /**
+     * Adds an additional exclusion to {@link #DEFAULT}.
+     * Does nothing if the default list has already been customized via {@link #FILE_OVERRIDE_LOCATION_PROPERTY}.
+     * @param filter a regular expression for {@link Class#getName} which, if matched according to {@link Matcher#matches}, will blacklist the class
+     */
+    public static void appendDefaultFilter(Pattern filter) {
```

Comment:
Throw checked exceptions to be able to propagate errors if we need to add some (like the class assumption failure below)